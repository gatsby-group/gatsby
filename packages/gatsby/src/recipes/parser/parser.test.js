const fs = require(`fs-extra`)
const path = require(`path`)

const parser = require(`.`)

const fixturePath = path.join(__dirname, `../prettier-git-hook.mdx`)
const fixtureSrc = fs.readFileSync(fixturePath, `utf8`)

test(`returns a set of commands`, async () => {
  const result = await parser.parse(fixtureSrc)

  expect(result.commands).toMatchSnapshot()
})

test(`partitions the MDX into steps`, async () => {
  const result = await parser.parse(fixtureSrc)

  expect(result.stepsAsMdx).toMatchSnapshot()
})

test(`handles imports from urls`, async () => {
  const result = await parser.parse(`
import TestRecipe from 'https://gist.githubusercontent.com/johno/20503d2a2c80529096e60cd70260c9d8/raw/0145da93c17dcbf5d819a1ef3c97fa8713fad490/test-recipe.mdx'

# Here is an imported recipe from a url!

<TestRecipe />
`)

expect(result.stepsAsMdx).toMatchSnapshot()
})

test(`fetches MDX from a url`, async () => {
  const result = await parser('https://gist.githubusercontent.com/johno/20503d2a2c80529096e60cd70260c9d8/raw/0145da93c17dcbf5d819a1ef3c97fa8713fad490/test-recipe.mdx')

  expect(result.commands).toMatchSnapshot()
})
