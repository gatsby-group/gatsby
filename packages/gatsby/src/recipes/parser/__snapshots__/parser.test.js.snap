// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches MDX from a url 1`] = `
Array [
  Object {
    "NPMPackageJson": Array [
      Object {
        "name": "test-recipe",
        "value": Object {
          "hello": "world!",
        },
      },
    ],
  },
]
`;

exports[`handles imports from urls 1`] = `
Array [
  "# Here is an imported recipe from a url!

# Test recipe

Add a package.json config object.

<NPMPackageJson
  name=\\"test-recipe\\"
  value={{
    hello: 'world!'
  }}
/>
",
]
`;

exports[`partitions the MDX into steps 1`] = `
Array [
  "# Prettier Git Hook

Make sure all of your code is run through prettier when you commit it to git.
We achieve this by configuring prettier to run on git hooks using husky and
lint-staged.
",
  "Install packages.

<NPMPackage name=\\"husky\\" />
<NPMPackage name=\\"prettier\\" />
<NPMPackage name=\\"lint-staged\\" />
",
  "Implement git hooks for prettier.

<NPMPackageJson
  name=\\"husky\\"
  value={{
    \\"hooks\\": {
      \\"pre-commit\\": \\"lint-staged\\"
    }
  }}
/>
<NPMPackageJson
  name=\\"lint-staged\\"
  value={{
    \\"*.{js,md,mdx,json}\\": [
      \\"prettier --write\\"
    ]
  }}
/>
",
  "Write prettier config files.

<File
  path=\\".prettierrc\\"
  content={\`{
  \\"semi\\": false,
  \\"singleQuote\\": true,
  \\"trailingComma\\": \\"none\\"
}\`}
/>
<File
  path=\\".prettierignore\\"
  content={\`.cache
public
node_modules
\`}
/>
",
  "Prettier, husky, and lint-staged are now installed! You can edit your \`.prettierrc\`
if you'd like to change your prettier configuration.
",
]
`;

exports[`returns a set of commands 1`] = `
Array [
  Object {},
  Object {
    "NPMPackage": Array [
      Object {
        "name": "husky",
      },
      Object {
        "name": "prettier",
      },
      Object {
        "name": "lint-staged",
      },
    ],
  },
  Object {
    "NPMPackageJson": Array [
      Object {
        "name": "husky",
        "value": Object {
          "hooks": Object {
            "pre-commit": "lint-staged",
          },
        },
      },
      Object {
        "name": "lint-staged",
        "value": Object {
          "*.{js,md,mdx,json}": Array [
            "prettier --write",
          ],
        },
      },
    ],
  },
  Object {
    "File": Array [
      Object {
        "content": "{
  \\"semi\\": false,
  \\"singleQuote\\": true,
  \\"trailingComma\\": \\"none\\"
}",
        "path": ".prettierrc",
      },
      Object {
        "content": ".cache
public
node_modules
",
        "path": ".prettierignore",
      },
    ],
  },
  Object {},
]
`;
